### PR TITLE
fix: add overflow property in text-widget

### LINKED ISSUE #10703: [Enhancement]: Enable scroll and Truncate text don't need separate switches in a text widget
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

When truncate text option enabled in text widget, Scroll switch is not valid anymore as we truncate. We can indeed club the option together instead.

Bad suggestion but nevertheless a suggestion : Create a dropdown called overflow text and add Scroll and Truncate as options there.

Screenshot 2022-01-28 at 11 30 18 AM
 

### Why should this be worked on?

Showing both options together doesn't make sense as one is dependent on other. @momcilo-appsmith can you please take a look at this?

### PR DESCRIPTION
**Pull request by techbhavin**
## Description

When truncating text option is enabled in a text widget, the Scroll switch is not valid anymore as we truncate to fixed this issue we made the following changes 

 - Add an `Overflow` Property Option in TextWidget with the following options 
      1. Scroll contents - Enables scroll in the text widget
      2. Truncate text - Enables truncation in the text widget
      3. No overflow - No scroll or truncation

Fixes #10703

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: enhancement/text-widget 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.78 **(0.02)** | 37.2 **(0.04)** | 35.36 **(0.01)** | 56.04 **(0.02)**
 :green_circle: | app/client/src/utils/DSLMigrations.ts | 73.25 **(0.11)** | 69.21 **(-0.06)** | 71.13 **(0)** | 73.16 **(0.11)**
 :sparkles: :new: | **app/client/src/utils/migrations/TextWidget.ts** | **100** | **92.86** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/TextWidget/constants.ts** | **100** | **100** | **100** | **100**
 :green_circle: | app/client/src/widgets/TextWidget/component/index.tsx | 73.08 **(0)** | 45.7 **(0.67)** | 57.69 **(0)** | 72.37 **(0)**
 :x: | ~~app/client/src/utils/migrations/TextWidgetReplaceTextStyle.ts~~ | ~~100~~ | ~~92.86~~ | ~~100~~ | ~~100~~</details>

### CODE REVIEW

File: app/client/src/widgets/TextWidget/component/index.tsx [hunk 1]
```diff
@@ -104,18 +104,27 @@ export const StyledText = styled(Text)<{
...
 }>`
   height: ${(props) =>
-    props.isTruncated ? `calc(100% - ${ELLIPSIS_HEIGHT}px)` : "100%"};
+    props.overflow === OverflowTypes.TRUNCATE
+      ? `calc(100% - ${ELLIPSIS_HEIGHT}px)`
+      : "100%"};
   overflow-x: hidden;
   overflow-y: ${(props) =>
-    props.scroll ? (props.isTruncated ? "hidden" : "auto") : "hidden"};
+    props.overflow === OverflowTypes.SCROLL
```
Comment (jsartisan) [inline app/client/src/widgets/TextWidget/component/index.tsx:112]: Will it be wise to switch statements here? multi-level ternary operator just makes my mind go numb. Your call.

**Review by jsartisan — APPROVED** [review]

**Comment by YogeshJayaseelan** [comment]
@techbhavin Tested this PR and it is working as expected

**Pull request merged**